crates/ra_hir/src/nameres/collector.rs

```diff
@@ -371,9 +371,18 @@ where
 
             if resolved_res.reached_fixedpoint != ReachedFixedPoint::Yes {
                 let crate_name = &path.segments[0].name;
+
+                // FIXME:
+                // $crate are not handled in resolver right now
+                if crate_name.to_string() == "$crate" {
+                    return true;
+                }
+
                 // FIXME:
```

Comment:
@matklad Do you think this comment is correct ?

Reply:
I must say I can't understand the example you are talking about. Could show the Rust code that causes problems? The best thing would be to add it as a test, actually.

We don't handle `#[cfg(test)]` right now, it is true. Ideally, the DefCollector should somehow "deactivate" all false cfg's, and we should be creating duplicates instances of the same Crate with different cfg flags.


However, for `#[cfg(test)]` specifically, I think we should just assume it is set. So, we should include dev-depdendencies into our crate graph by default. If this doesn't happen, then we should fix our project model

Reply:
>  So, we should include dev-depdendencies into our crate graph by default. If this doesn't happen, then we should fix our project model

Yes, it seem to be a bug in CargoWorkspace. I filed #1282 for that.